### DISCUSSION TITLE
Trouble with "expected a pointer"

### DISCUSSION THREAD

**Discussion by IcarusTwine** [vfsfitvnm/frida-il2cpp-bridge#331] [Q&A]
Hi, Currently i'm accessing a live instance of a class due to the method not being static, 
This is my current code: 
```js
            const UIMgr =
              Il2Cpp.Domain.assemblies["MainThread"].image.classes[
                "UIModule.M.UIMgr"
              ];
            let garbage = Il2Cpp.GC.choose(UIMgr);
            garbage.forEach((instance) => {
              instance.methods.OpenPage.invoke(
                Il2Cpp.String.from("pref_BattlePromptPage"),
                null, // UIPageData
                true
              );
            });
 ````
 Using a trace for the function when it happens i see:
 ```
 [il2cpp] 0x0362ad78 ┌─UIModule.M.UIMgr.OpenPage(this = GameRoot (UIModule.M.UIMgr), System.String pageName = "pref_BattlePromptPage", UIModule.M.UIPageData data = UIModule.M.UIPageData, System.Boolean focus = true)
 ```
When placing in UIPageData simular to my const of UIMgr I get either a crash or 
```
Error: expected a pointer
    at <anonymous> (node_modules/frida-il2cpp-bridge/dist/il2cpp/utils.js:181)
```

I'm specifically trying to find how I correctly place `UIModule.M.UIPageData data = UIModule.M.UIPageData` into the null part of my code. 
I'm still learning frida so many thanks in advance.

**Comment by vfsfitvnm** [comment]
Hi, if you want to pass a `null`, you should use `NULL` instead. Alternatively, `new Il2Cpp.Object(NULL)`
